### PR TITLE
Optimize transitions

### PR DESCRIPTION
**Pull request by daun**
Based on work by @hirasso in #15 

- Ensure value never exceeds 1
- Initialise with random value
- Improve initial transition by forcing reflow
- Some refactoring and additional parameters

### CODE REVIEW

**Comment by hirasso** [comment]
Hi @daun , any news on when you want to merge this into `master`? :)

**Comment by daun** [comment]
@hirasso Ready to merge, just waiting for a review by another maintainer.

**Review by gmrchk — APPROVED** [review]

**Comment by gmrchk** [comment]
Apologies didn't notice I was blocking!

**Pull request merged**
